### PR TITLE
 [MLIR][Vector] Allow Scalable Dim in OneDimMultiReductionToTwoDim

### PR DESCRIPTION
**Pull request by zhaoshiz**
To correctly lower multi_reduction of 1-dim scalable vector, e.g.: <[4]xf32>

### CODE REVIEW

File: mlir/test/Dialect/Vector/vector-multi-reduction-lowering.mlir [hunk 1]
```diff
@@ -281,6 +282,23 @@ func.func private @scalable_dims(%A : vector<8x[4]x2xf32>, %B: vector<8x[4]xf32>
 // CHECK:           %[[VAL_163:.*]] = vector.shape_cast %[[VAL_162]] : vector<[32]xf32> to vector<8x[4]xf32>
 // CHECK:           return %[[VAL_163]] : vector<8x[4]xf32>
 
+// Check that OneDimMultiReductionToTwoDim handles scalable dim
+func.func private @scalable_dim_1d(%A: vector<[4]xf32>, %B: f32, %C: vector<[4]xi1>) -> f32 {
```
Comment (c-rhodes) [inline mlir/test/Dialect/Vector/vector-multi-reduction-lowering.mlir:286]: nit: private can be dropped
```suggestion
func.func @scalable_dim_1d(%A: vector<[4]xf32>, %B: f32, %C: vector<[4]xi1>) -> f32 {
```

File: mlir/test/Dialect/Vector/vector-multi-reduction-lowering.mlir [hunk 2]
```diff
@@ -253,6 +253,7 @@ func.func private @scalable_dims(%A : vector<8x[4]x2xf32>, %B: vector<8x[4]xf32>
   %0 = vector.multi_reduction <add>, %A, %B [2] : vector<8x[4]x2xf32> to vector<8x[4]xf32>
   return %0 : vector<8x[4]xf32>
 }
+
```
Comment (c-rhodes) [inline mlir/test/Dialect/Vector/vector-multi-reduction-lowering.mlir:256]: nit: unrelated change

**Review by c-rhodes — APPROVED** [review]
couple of minor nits but otherwise LGTM, cheers!

File: mlir/test/Dialect/Vector/vector-multi-reduction-lowering.mlir (hunk 1 above)
Comment (zhaoshiz) [reply mlir/test/Dialect/Vector/vector-multi-reduction-lowering.mlir]: done
